### PR TITLE
Fix issue with Batch Account update

### PR DESCRIPTION
**Pull request by matthchr**
- KeyVaultReference and PoolAllocationMode are not updatable

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by matthchr** [comment]
@markcowl - Since we unlisted 4.1 and are going to leave it unlisted, even though this is a breaking change I am going to leave it versioned 4.2 because it will be the first visible nuget package in the 4.* family.

**Comment by matthchr** [comment]
Also not sure why CLA is required.

File: src/ResourceManagement/Batch/Microsoft.Azure.Management.Batch/Generated/Models/BatchAccountUpdateParameters.cs [hunk 1]
```diff
@@ -38,16 +38,10 @@ public partial class BatchAccountUpdateParameters
         /// account.</param>
         /// <param name="autoStorage">The properties related to auto storage
         /// account.</param>
-        /// <param name="poolAllocationMode">The allocation mode to use for
-        /// creating pools in the Batch account.</param>
-        /// <param name="keyVaultReference">A reference to the Azure key vault
-        /// associated with the Batch account.</param>
-        public BatchAccountUpdateParameters(IDictionary<string, string> tags = default(IDictionary<string, string>), AutoStorageBaseProperties autoStorage = default(AutoStorageBaseProperties), PoolAllocationMode? poolAllocationMode = default(PoolAllocationMode?), KeyVaultReference keyVaultReference = default(KeyVaultReference))
+        public BatchAccountUpdateParameters(IDictionary<string, string> tags = default(IDictionary<string, string>), AutoStorageBaseProperties autoStorage = default(AutoStorageBaseProperties))
```
Comment (markcowl) [inline src/ResourceManagement/Batch/Microsoft.Azure.Management.Batch/Generated/Models/BatchAccountUpdateParameters.cs]: Need to update generate.cmd to refer to the spec using the commit for the recently merged PR.  Othewrwise this looks good

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by matthchr** [comment]
@markcowl We actually don't have a generate.cmd, we have a RegenerateBatch.cmd and it doesn't point to a commit hash it requires a specific file path to be given.

Do you want me to fix this in this PR or keep things as is and fix later when there isn't a deadline?
Thanks,
-Matt

**Comment by matthchr** [comment]
Actually I will fix it, it's an easy change.

**Review by markcowl — APPROVED** [review]

**Comment by matthchr** [comment]
@markcowl Tests pass, can we merge this?

**Pull request merged**